Documentation/Relationship/Relationship.md

```diff
@@ -463,7 +463,7 @@ struct TestService: WebService {
     Group {
       "user"
       $userId
-        .relationship(to: Greeting.self, name: "greeter")
+        .relationship(name: "greeter", to: Greeting.self)
```

Comment:
I will revise this use case in more detailed again, but this should be supported out of the box.
The relationship modifier would create a `PartialRelationshipCandidate`. When it is later resolved (after the inheritance step; calling `ensureResolved` in the `TypeIndex`), resolvers for all path parameters (or property resolvers leading to the inherited `"self"`) are added for any explicit relationship definition.

The only thing what is not possible with `.relationship(name:to:)` is adding additional resolvers for additional path parameters. I had this in a prototype, but removed it again, since you could add such a modifier anywhere. This would be a problem, if you would add a resolver for a property (like what `References` does), but the property is then later eliminated by a response transformer.

Reply:
> I will revise this use case in more detailed again, but this should be supported out of the box. 

Sounds great 🥳

> The only thing what is not possible with .relationship(name:to:) is adding additional resolvers for additional path parameters. I had this in a prototype, but removed it again, since you could add such a modifier anywhere. This would be a problem, if you would add a resolver for a property (like what References does), but the property is then later eliminated by a response transformer.

I am not 100% sure if I understood this correctly. You are saying that if the greeter would be located behind an additional path parameter it would be impossible to reference it would be impossible to reference it? Or am I on the wrong track here?

Reply:
It is still possible to reference/link to it. We simply won't have a value for it when formatting. So we would just put `../{pathParamName}/..` into the `_links` section instead of a dedicated value (and the requester has to provide a value then if they want to create a request against that linked endpoint).
Does this bring clarification. Otherwise I can just state an extensive example.

Reply:
To give some more context. 

`RelationshipSource` (or `Relationship` when in the `WithRelationships` namespace) is basically the DSL equivalent for the `.relationship(name:to:)` modifier. The `RelationshipSource` does indeed support additional PathParameterResolvers (e.g. using values from propreties). But the `RelationshipSource` also doesn't require that all path parameters to the destination are resolvable (=> meaning they appear as described above without a value in the `_links` section).

`RelationshipReference` (or `References`) on the other hand requires that all path parameters are resolved to the destination.

Reply:
Thanks for the detailed explanation, now I understand 👍